### ISSUE TITLE
Support for XFS

### ISSUE THREAD

**Issue by dndusdndus12** [sleuthkit/sleuthkit#1465]
We implemented support for XFS, a file system that TSK does not support.

Related requests can also be found in #1365 , and development progress is #1461 in a pull request state.

We are now correcting the link error in the MSVC environment and will be completed soon.

**Comment by PracaGrande** [comment]
Hi, this is great!
Wanted to test but seems the functionality is still not committed to the development branch, correct?

**Comment by dndusdndus12** [comment]
Yes correct. 
It supports the basic features of tsk recover with the -e option (file extraction) and fsstat for the latest version XFS file system installed through xfsprogs, and is still developing additional features and more file system support(CentOS, RHEL). 
We will update via pull request whenever it is stabilized through internal testing.

**Comment by PracaGrande** [comment]
Thank you!
 Is the support for the fls utility also scheduled for this first release?

**Comment by dndusdndus12** [comment]
Yes, because the function 'fls' uses 'inode walk' and 'load attr' that we already implemented. 
Since we have followed the development philosophy of tsk, the features supported by this version are similar to those supported by other file systems.

Here are the results we just tested.
![test](https://user-images.githubusercontent.com/3907683/55670501-12917b00-58c0-11e9-9c74-5f970043de06.png)

**Comment by PracaGrande** [comment]
Great, I currently doing some forensic work on a Linux Red Hat 7.3 OS with a XFS and would love to try it out. Also to see if it would support obtaining metadata about deleted files.

**Comment by dndusdndus12** [comment]
There is another pull request for XFS support. You might want to check it(#1476).

**Comment by isciurus** [comment]
@PracaGrande as long as you have an inode number, and that inode is not overwritten, you can get info with ils/icat/istat of my #1476, or do fls starting from that inode. I believe #1476 was tested thoroughly on a few images, feel free to try it on your image and report bugs. Fls is also fully supported in my fork.

Also, @dndusdndus12, is your fork under active development too? Do we want to plan working on a single implementation jointly instead of ending up with two different codebases?

**Comment by dndusdndus12** [comment]
@isciurus 
My research team is developing various digital forensic topics including XFS file system.
Also, I'm wondering about your development status. Are you talking with a TSK contributor?
As you know, this is a time-consuming task because the development code targets various versions of XFS and the scope of development is not consistent.
It will be better to communicate by email with detail issues like collaboration, schedule, and so on.

**Comment by PracaGrande** [comment]
Hello, wondering if there is any update on TSK support for XFS?

**Comment by malaclyps3** [comment]
Hello, I am also wondering what happend to the sleuthkit XFS support?

**Comment by angry-bender** [comment]
Just compiled this one today, as it seems to process inodes properly

https://github.com/isciurus/sleuthkit

Definitely needs a PR @isciurus if you're keen? This issue has been open a while 😊, and from my testing I think your implementation seems to be working best. @jayaramcs just wanted to mention you here as a maintainer, as this one's been outstanding for a while.

I think there is also a couple of duplicate issues here such as #1465

**Comment by angry-bender** [comment]
@markmckinnon sorry to bother again, but I think this is a solid PR 😊

**Comment by isciurus** [comment]
@angry-bender thanks. The PR was filed under https://github.com/sleuthkit/sleuthkit/pull/1476 (plus there is also https://github.com/sleuthkit/sleuthkit/pull/1461). It's not actively supported at the moment, so I didn't find time to address all the issues from https://github.com/sleuthkit/sleuthkit/pull/1476#issuecomment-643473905. If anyone wants to take a closer look and maybe fix some of them, that would certainly help.
